src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -91,7 +91,7 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
   override def upsert(rows: Seq[Row]): IO[Unit] = {
     val assets = rows.map(fromRow[AssetsUpsertSchema](_))
     val (itemsToUpdate, itemsToUpdateOrCreate) =
-      assets.partition(r => r.id.exists(_ > 0) || (r.name.isEmpty && r.externalId.nonEmpty))
+      assets.partition(r => r.id.exists(_ > 0) || (r.name.isEmpty && r.getExternalId().nonEmpty))
```

Comment:
Any reason for the preference on `getExternalId()` over `externalId` here?

Reply:
Because externalId contains the `OptionalField` while getExternalId is converted to Option